pipeline/docker/__init__.py

```diff
@@ -110,5 +129,26 @@ def create_docker_compose(path, **environment_vars):
             },
         },
     }
+
+    if gpu_index is not None:
+        docker_yml_dict["services"]["pipeline_api"]["deploy"] = {
+            "resources": {
+                "reservations": {
+                    "devices": [
+                        {
+                            "driver": "nvidia",
+                            "device_ids": ["0"],
+                            "capabilities": ["gpu"],
+                        }
+                    ]
+                }
+            }
+        }
+
     with open(os.path.join(path, "docker-compose.yml"), "w") as docker_compose_file:
         docker_compose_file.write(yaml.dump(docker_yml_dict))
+
+
+def build():
+    subprocess.run("sudo docker compose pull".split(" "))
+    subprocess.run("sudo docker compose build".split(" "))
```

Comment:
The default behaviour of `str.split` is to split on whitespace, so you could omit the argument here.